OrbitCore/SamplingProfiler.cpp

```diff
@@ -344,48 +354,15 @@ void SamplingProfiler::FillThreadSampleDataSampleReports() {
             100.f * it->second / threadSampleData.m_NumSamples;
       }
       function.m_Address = address;
-
-      std::shared_ptr<Module> module = m_Process->GetModuleFromAddress(address);
-      function.m_Module = module ? module->m_Name : "???";
+      function.m_Module = Capture::GAddressToModuleName[address];
```

Comment:
I am always somewhat hesitant to use the std containers in this way, since [] will return the default constructed value if the key does not exist. I guess it is fine how it is currently done, but you could also add a CHECK if you are sure that this key always exists, or if you are unsure, maybe test and use "???" as before

Reply:
Basically, I followed the logic that we have for Capture::GAddressToFunctionName: we fill map in UpdateAddressInfo() (with "???" if we don't have an information and with function/module name if we have) and then we use this map in other functions. In both functions (UpdateAddressInfo and FillThreadSampleDataSampleReports) we get information about addresses from the one place (m_UniqueCallstacks), so we shouldn't be in the situation when [] calls the default constructor.

Reply:
Should we use "at" instead of "[]" for this case then?

Reply:
at() can throw exceptions, I would prefer having
`CHECK(Capture::GAddressToModuleName.count(address) != 0);`

Reply:
I would also prefer `at` here.

Reply:
The general question is "When do we prefer to have an exception and when an abort()"?